applications/FSIapplication/custom_utilities/partitioned_fsi_utilities.hpp

```diff
@@ -247,94 +215,245 @@ class PartitionedFSIUtilities
+            ux_mesh_norm += std::pow(it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT_X), 2);
+            uy_mesh_norm += std::pow(it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT_Y), 2);
+            uz_mesh_norm += std::pow(it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT_Z), 2);
+        }
 
-    /*@} */
-    /**@name Protected Operators*/
-    /*@{ */
+        rInterfaceModelPart.GetCommunicator().SumAll(p_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(vx_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(vy_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(vz_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(rx_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(ry_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(rz_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(ux_mesh_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(uy_mesh_norm);
+        rInterfaceModelPart.GetCommunicator().SumAll(uz_mesh_norm);
+
+        if (rInterfaceModelPart.GetCommunicator().MyPID() == 0)
+        {
+            std::cout << " " << std::endl;
+            std::cout << "|p_norm| = " << std::sqrt(p_norm) << std::endl;
+            std::cout << "|vx_norm| = " << std::sqrt(vx_norm) << std::endl;
+            std::cout << "|vy_norm| = " << std::sqrt(vy_norm) << std::endl;
+            std::cout << "|vz_norm| = " << std::sqrt(vz_norm) << std::endl;
+            std::cout << "|rx_norm| = " << std::sqrt(rx_norm) << std::endl;
+            std::cout << "|ry_norm| = " << std::sqrt(ry_norm) << std::endl;
+            std::cout << "|rz_norm| = " << std::sqrt(rz_norm) << std::endl;
+            std::cout << "|ux_mesh_norm| = " << std::sqrt(ux_mesh_norm) << std::endl;
+            std::cout << "|uy_mesh_norm| = " << std::sqrt(uy_mesh_norm) << std::endl;
+            std::cout << "|uz_mesh_norm| = " << std::sqrt(uz_mesh_norm) << std::endl;
+            std::cout << " " << std::endl;
+        }
+    }
 
+    /**
+     * Computes and prints the structure interface residual norms for debugging purposes
+     * @param rInterfaceModelPart: interface modelpart in where the residual is computed
+     */
+    virtual void ComputeAndPrintStructureInterfaceNorms(ModelPart& rInterfaceModelPart)
+    {
+        double ux_norm = 0.0;
+        double uy_norm = 0.0;
+        double uz_norm = 0.0;
+
+        auto& rLocalMesh = rInterfaceModelPart.GetCommunicator().LocalMesh();
+        ModelPart::NodeIterator local_mesh_nodes_begin = rLocalMesh.NodesBegin();
+        #pragma omp parallel for firstprivate(local_mesh_nodes_begin) reduction(+ : ux_norm, uy_norm, uz_norm)
+        for(int k=0; k<static_cast<int>(rLocalMesh.NumberOfNodes()); ++k)
+        {
+            const ModelPart::NodeIterator it_node = local_mesh_nodes_begin+k;
 
-    /*@} */
-    /**@name Protected Operations*/
-    /*@{ */
+            ux_norm += std::pow(it_node->FastGetSolutionStepValue(DISPLACEMENT_X), 2);
```

Comment:
the procedure you used accesses 3 times the database.
inestead it would be better to do

    const array_1d<double,3>& d = it_node->FastGetSolutionStepValue(DISPLACEMENT)
    ux_norm += d[0]*d[0];
    uy_norm += d[1]*d[1];
    ...

Reply:
Done!